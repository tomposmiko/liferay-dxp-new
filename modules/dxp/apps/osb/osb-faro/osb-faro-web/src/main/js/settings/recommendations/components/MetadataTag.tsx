import Label from 'shared/components/Label';
import React from 'react';

const MetadataTag: React.FC<{value: string}> = ({value}) => (
	<Label className='metadata-tag-root' size='lg'>
		{value}
	</Label>
);

export default MetadataTag;
