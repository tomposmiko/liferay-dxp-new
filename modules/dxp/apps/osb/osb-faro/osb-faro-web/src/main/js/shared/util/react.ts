import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
