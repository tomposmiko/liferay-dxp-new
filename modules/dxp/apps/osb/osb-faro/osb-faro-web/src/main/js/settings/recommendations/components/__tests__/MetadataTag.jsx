import MetadataTag from '../MetadataTag';
import React from 'react';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('MetadataTag', () => {
	it('should render', () => {
		const {container} = render(<MetadataTag value='og:url' />);

		expect(container).toMatchSnapshot();
	});
});
