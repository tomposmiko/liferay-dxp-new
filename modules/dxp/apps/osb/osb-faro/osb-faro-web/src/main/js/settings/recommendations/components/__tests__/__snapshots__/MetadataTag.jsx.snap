// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetadataTag should render 1`] = `
<div>
  <div
    class="label label-root metadata-tag-root label-lg"
  >
    <span
      class="label-item"
    >
      og:url
    </span>
  </div>
</div>
`;
